Libraries/Image/RCTImageLoader.m

```diff
@@ -370,12 +370,28 @@ - (RCTImageLoaderCancellationBlock)loadImageOrDataWithTag:(NSString *)imageTag
 
     // Check for cached response before reloading
     // TODO: move URL cache out of RCTImageLoader into its own module
-    NSCachedURLResponse *cachedResponse = [_URLCache cachedResponseForRequest:request];
-    if (cachedResponse) {
-      processResponse(cachedResponse.response, cachedResponse.data, nil);
-      return;
+    while(true) {
```

Comment:
`while (true)` is a big code smell. I'd prefer it if we repeated the `NSCachedURLResponse *cachedResponse = [_URLCache cachedResponseForRequest:request];` line instead and use that as the loop condition.